### PR TITLE
Create samples for C# 6 that leverage the interactive window.

### PR DESCRIPTION
**Pull request by BillWagner**
Add new samples for object and collection initializers. These are for dotnet/docs#5169

Create samples for the new interactive "Explore C# 6" tutorial. This contributes to dotnet/docs#8206

### CODE REVIEW

**Comment by Thraka** [comment]
If these are written for C# 6, how do we make sure that the snippet isn't updated in the future and uses a feature from C# 7 or 8? It might be good to put the `csharp6` in the file path or something?

**Comment by BillWagner** [comment]
Good catch @Thraka 

I meant to add the `<LangVersion>6.0</LangVersion>` in the csproj file.

**Comment by Thraka** [comment]
Whoops, I should have said, I meant the files like: `classes-and-structs/object-collection-initializers/BasicObjectInitializers.cs` The path to these files is generic and likely represents the latest-and-greatest ? So having code in those files for the language specific articles could be troublesome in the future.

File: snippets/csharp/new-in-6/initializers.cs [hunk 1]
```diff
@@ -5,11 +5,11 @@ namespace Initializers
     public class Examples
     {
         // <ListInitializer>
```
Comment (Thraka) [inline snippets/csharp/new-in-6/initializers.cs]: This snippet tag says ListInitializer bu t it's using an IndexInitializer ?

Comment (BillWagner) [reply snippets/csharp/new-in-6/initializers.cs]: Changed here, and in the use in dotnet/docs#9551

**Pull request merged**